pyproject.toml

```diff
@@ -30,9 +30,8 @@ dependencies = [
     # scipy is currently bounded from above because scipy 1.11 removed the
     # `sym_pos` kwarg from `scipy.linalg.solve`, but pyscf has not yet been updated.
     "scipy>=1.5.2,<1.11",
-    "rustworkx>=0.12.0",
     "qiskit-aer>=0.12.0",
-    "qiskit-terra>=0.24.0",
+    "qiskit>=0.44.0",
```

Comment:
This actually bumps the version of terra, because Qiskit 0.44 == qiskit-terra 0.25.  I think this is worth doing, because it avoid adding dependencies on qiskit-aer and qiskit-ibmq-provider.  But it does mean we should add a brief release note, and edit the version number in the badge in the README.

Reply:
OK sounds good, I do think it's the right call to let the meta-package control everything. I believe this is what the Qiskit devs are hoping clients will do sooner than later

Reply:
That's consistent with the message I was trying to convey.

But actually, I've changed my mind.  Why not just change to `qiskit>=0.43.0`?  Then we don't need a release note, and people installing a newer verson of qiskit (everyone except the "minimum version tests") can still get by without ibmq-provider.  (We can also avoid bumping rustworkx, not that it really matters.)

My earlier concern was that the minimum version tests wouldn't figure this out, but since the metapackage pins the version of terra precisely, it won't be a problem.